pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/DefaultNullValueVirtualColumnProvider.java

```diff
@@ -85,7 +85,13 @@ public InvertedIndexReader<?> buildInvertedIndex(VirtualColumnContext context) {
 
   @Override
   public ColumnMetadataImpl buildMetadata(VirtualColumnContext context) {
-    return getColumnMetadataBuilder(context).setCardinality(1).setSorted(context.getFieldSpec().isSingleValueField())
-        .setHasDictionary(true).build();
+    if (context.getFieldSpec().isSingleValueField()) {
+      return getColumnMetadataBuilder(context).setCardinality(1).setSorted(true)
+          .setTotalDocs(context.getTotalDocCount())
+          .setHasDictionary(true).build();
+    }
+    return getColumnMetadataBuilder(context).setSorted(false).setTotalDocs(context.getTotalDocCount())
+        .setHasDictionary(true).setMaxNumberOfMultiValues(0)
+        .build();
```

Comment:
It should be set to 1
```suggestion

    ColumnMetadataImpl.Builder builder = getColumnMetadataBuilder(context).setCardinality(1).setHasDictionary(true);
    if (context.getFieldSpec().isSingleValueField()) {
      builder.setSorted(true);
    } else {
      builder.setMaxNumberOfMultiValues(1);
    }
    return builder.build();
```